server/util.go

```diff
@@ -136,15 +136,30 @@ func externalIP() (string, error) {
 			case *net.IPAddr:
 				ip = v.IP
 			}
-			if ip == nil || ip.IsLoopback() {
-				continue
+			if external_ip := isPublicIP(ip); ip != nil {
```

Comment:
I think we can remove the above switch-case in favor of this function

Reply:
I'm not sure how we can get rid of the switch case, other than putting it in the isPublicIp method? Is that what you mean?

Reply:
Or at least deduplicating the checks - at the moment, we check both whether the interface and whether the IP is a loopback, which I think is unnecessary?